src/test/java/org/jabref/logic/citationstyle/CitationStyleGeneratorTest.java

```diff
@@ -33,113 +41,176 @@ void testIgnoreCarriageReturnNewLine() {
+        String style = CitationStyle.getDefault().getSource();
+
+        String actualCitation = CitationStyleGenerator.generateCitation(firstEntry, style, CitationStyleOutputFormat.TEXT, bibDatabaseContext, bibEntryTypesManager);
+        assertEquals(expectedCitation, actualCitation);
+    }
+
+    @Test
+    void testBibtexWithNumber() {
+        BibEntry entry = new BibEntry(StandardEntryType.Article);
+        entry.setField(StandardField.AUTHOR, "Last, First and\nDoe, Jane");
+        entry.setField(StandardField.NUMBER, "7");
+
+        // if the default citation style changes this has to be modified
+        String expected = "[1]F. Last and J. Doe, Art. no. 7.\n";
+
+        BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(new BibDatabase(List.of(entry)));
+        bibDatabaseContext.setMode(BibDatabaseMode.BIBTEX);
+
+        String citation = CitationStyleGenerator.generateCitation(entry, CitationStyle.getDefault().getSource(), CitationStyleOutputFormat.TEXT, bibDatabaseContext, bibEntryTypesManager);
+        assertEquals(expected, citation);
+    }
+
+    @Test
+    void testBiblatexWithIssue() {
+        BibEntry entry = new BibEntry(StandardEntryType.Article);
+        entry.setField(StandardField.AUTHOR, "Last, First and\nDoe, Jane");
+        entry.setField(StandardField.ISSUE, "7");
+
+        // if the default citation style changes this has to be modified
+        String expected = "[1]F. Last and J. Doe, no. 7.\n";
+
+        BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(new BibDatabase(List.of(entry)));
+        bibDatabaseContext.setMode(BibDatabaseMode.BIBLATEX);
+
+        String citation = CitationStyleGenerator.generateCitation(entry, CitationStyle.getDefault().getSource(), CitationStyleOutputFormat.TEXT, bibDatabaseContext, bibEntryTypesManager);
+        assertEquals(expected, citation);
+    }
+
+    @Test
+    void testBiblatexWitNumber() {
+        BibEntry entry = new BibEntry(StandardEntryType.Article);
+        entry.setField(StandardField.AUTHOR, "Last, First and Doe, Jane");
+        entry.setField(StandardField.NUMBER, "7");
+
+        // if the default citation style changes this has to be modified
+        String expected = "[1]F. Last and J. Doe, Art. no. 7.\n";
+
+        BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(new BibDatabase(List.of(entry)));
+        bibDatabaseContext.setMode(BibDatabaseMode.BIBLATEX);
+
+        String citation = CitationStyleGenerator.generateCitation(entry, CitationStyle.getDefault().getSource(), CitationStyleOutputFormat.TEXT, bibDatabaseContext, bibEntryTypesManager);
+        assertEquals(expected, citation);
+    }
+
+    @Test
+    void testBiblatexWithIssueAndNumber() {
+        BibEntry entry = new BibEntry(StandardEntryType.Article);
+        entry.setField(StandardField.AUTHOR, "Last, First and\nDoe, Jane");
+        entry.setField(StandardField.ISSUE, "7");
+        entry.setField(StandardField.ISSUE, "28");
```

Comment:
Is this a typo? Issue x2. Above it says "withIssueAndNumber".

Reply:
Oh thanks for spotting! This was not intended.

Reply:
As you can now see at the tests, the bibtex number gets now rendered as Art no.